src/frontend/admin/src/contexts/auth/AuthProvider.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from "react";
+import { PropsWithChildren, useEffect, useMemo, useState } from "react";
+import { AuthContext, AuthContextInterface } from "./AuthContext";
+import { AuthenticatedUser } from "@/types/auth";
+import { Maybe } from "@/types/utils";
+import { AuthRepository } from "@/services/repositories/auth/AuthRepository";
+import { PATH_ADMIN } from "@/utils/routes/path";
+
+export function AuthProvider({ children }: PropsWithChildren) {
+  const [currentUser, setCurrentUser] =
+    useState<Maybe<AuthenticatedUser>>(undefined);
+
+  useEffect(() => {
+    AuthRepository.me().then((user) => {
+      setCurrentUser(user);
+    });
+  }, []);
+
+  const authContext: AuthContextInterface = useMemo(
+    () => ({
+      currentUser,
+      logout: () => (window.location.href = PATH_ADMIN.auth.logout()),
```

Comment:
This is only used in `DashboardLayoutNavAccount` l.102
It is confusing to have both `authContext.logout()` and `PATH_ADMIN.auth.logout()`
We may remove this method and directly use `window.location.href = PATH_ADMIN.auth.logout())` in `DashboardLayoutNavAccount`

Reply:
Indeed for a first intent, this complexity is useless.